### ISSUE TITLE
Body missing in request

### ISSUE THREAD

**Issue by kirkbauer2** [XeroAPI/xero-php-oauth2#165]
**SDK you're using (please complete the following information):**
2.1.2 (latest)

**Describe the bug**
I'm trying to create a CreditNote, but the actual PUT request to the Xero API has no body.

**To Reproduce**
Here is my code:

        $result = $xero->createCreditNotes( XERO_TENANT_ID, $outgoing_tokens, $summarize_errors, $unitdp );

I added debugging on line 5790 of lib/Api/AccountingApi.php and confirmed that the Request object is being created with a valid string for its body:

        echo( "GOT HERE: " . $httpBody . "\n" );

Here is what I see on the console, starting with my debugging statement:

GOT HERE: {"creditNotes":[{"type":"ACCPAYCREDIT","contact":{"contactID":"6c22a74c-c78f-436d-be35-17aacf8c4571"},"date":"2020-09-17","lineItems":[{"item_code":"2019-lightblue-potion-condense","description":"Potion Condensed Healing [2019 Lightblue]","quantity":1,"account_code":"1400","unit_amount":0.5},{"description":"one-time correction for Ed arrangement","quantity":1,"account_code":"0100","unit_amount":-0.5}]}]}
Exception when calling AccountingApi->createCreditNotes: [400] Client error: `PUT https://api.xero.com/api.xro/2.0/CreditNotes?summarizeErrors=false&unitdp=4` resulted in a `400 Bad Request` response:
{
  "ErrorNumber": 17,
  "Type": "NoDataProcessedException",
  "Message": "No data has been processed for this endpoi (truncated...)

I confirmed with mitmproxy that no body is being sent:

2020-09-17 00:12:50 PUT https://api.xero.com/api.xro/2.0/CreditNotes?summarizeErrors=false&unitdp=4
                        ← 400 Bad Request application/json 206b 2.22s
               Request                              Response                              Detail
Host:            api.xero.com
User-Agent:      [xero-php-oauth2 (2.1.2)]
xero-tenant-id:  19cbe07e-fd5d-4b35-be9c-561506d208e5
Accept:          application/json
Content-Type:    application/json
Authorization:   Bearer eyJhbGciOiJSUzI1NiIsImtpZCI6IjFDQUY4RTY2NzcyRDZEQzAy...

**Expected behavior**
The body of the PUT request should include the JSON structure of my new object.

**Screenshots**
I am attaching a screenshot showing (1) the debugging command I added to the library and (2) the mitmproxy window showing a missing body.

<img width="1870" alt="Untitled" src="https://user-images.githubusercontent.com/1523423/93419961-10389c00-f87c-11ea-8158-04c149bd96ee.png">

**Comment by SidneyAllen** [comment]
@kirkbauer2 - Looking at your createCreditNotes method, it's unclear what you are passing as your 2nd param $outgoing_tokens.

Here is a snippet of function code from our "kitchen sync" app available here.  
https://github.com/XeroAPI/xero-php-oauth2-app

    $contact = new XeroAPI\XeroPHP\Models\Accounting\Contact;

    $contact->setContactId($contactId);

    $arr_creditnotes = [];

    $creditnote_1 = new XeroAPI\XeroPHP\Models\Accounting\CreditNote;
    $creditnote_1->setDate(new DateTime('2019-12-15'))
	->setContact($contact)
	->setLineItems($lineitems)
	->setType(XeroAPI\XeroPHP\Models\Accounting\CreditNote::TYPE_ACCPAYCREDIT);
    array_push($arr_creditnotes, $creditnote_1);
	
    $creditnote_2 = new XeroAPI\XeroPHP\Models\Accounting\CreditNote;
    $creditnote_2->setDate(new DateTime('2019-12-02'))
	->setContact($contact)
	->setLineItems($lineitems)
	->setType(XeroAPI\XeroPHP\Models\Accounting\CreditNote::TYPE_ACCPAYCREDIT);
    array_push($arr_creditnotes, $creditnote_2);
			
    $creditnotes = new XeroAPI\XeroPHP\Models\Accounting\CreditNotes;
    $creditnotes->setCreditNotes($arr_creditnotes);

    $result = $apiInstance->createCreditNotes($xeroTenantId,$creditnotes);

**Comment by kirkbauer2** [comment]
Thank you for your response. I changed my code to completely follow the example you provided above. Unfortunately there was no change in behavior. But it is good to know about that method of building objects because I didn't see that in the other documentation. Nor was I aware of the "Kitchen Sync" example app.

But the good news is I figured out my problem. In my handling of the storage and retrieval of the auth token, I added a carriage return to the end. When I passed that in to XeroAPI\XeroPHP\Configuration::getDefaultConfiguration()->setAccessToken() it didn't have a problem with it. When it added it to the "Authorization" header in the HTTP request it added a blank line, which is the signal to the server that the headers are complete. Unfortunately the next header was Content-Length which was therefore ignored. The server assumed there was no message body, even though there was one.

My suggestion is to catch this and error, or otherwise strip off whitespace off of auth tokens passed in to the API.
